generator/transit_generator.cpp

```diff
@@ -43,20 +60,79 @@ string GetFileName(string const & filePath)
-/// but before serialization to mwm.
 template <class Item>
-void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer,
-                     function<void(vector<Item> &)> handler = nullptr)
+void DeserializeFromJson(my::Json const & root, string const & key, vector<Item> & items)
 {
-  vector<Item> items;
-
+  items.clear();
   DeserializerFromJson deserializer(root.get());
   deserializer(items, key.c_str());
+}
+
+void DeserializeGatesFromJson(my::Json const & root, string const & mwmDir, string const & countryId,
+                              vector<Gate> & gates)
+{
+  DeserializeFromJson(root, "gates", gates);
+
+  // Creating IndexRouter.
+  LocalCountryFile localCountryFile;
+  Index index;
+  auto const result = RegisterSingleMwm(
+      my::JoinFoldersToPath(mwmDir, countryId + DATA_FILE_EXTENSION), localCountryFile, index);
 
-  if (handler)
-    handler(items);
+  auto infoGetter = storage::CountryInfoReader::CreateCountryInfoReader(GetPlatform());
+  CHECK(infoGetter, ());
+
+  auto const countryFileGetter = [&infoGetter](m2::PointD const & pt) {
+    return infoGetter->GetRegionCountryId(pt);
+  };
+
+  auto const getMwmRectByName = [&](string const & c) -> m2::RectD {
+    CHECK_EQUAL(countryId, c, ());
+    return infoGetter->GetLimitRectForLeaf(c);
+  };
+
+  CountryFile const countryFile(countryId);
+  CHECK_EQUAL(result.first.GetInfo()->GetType(), MwmInfo::COUNTRY, ());
+  auto numMwmIds = make_shared<NumMwmIds>();
+  numMwmIds->RegisterFile(countryFile);
+
+  // Note. |indexRouter| is valid until |index| is valid.
+  IndexRouter indexRouter(VehicleType::Pedestrian, false /* load altitudes */,
+                          CountryParentNameGetterFn(), countryFileGetter, getMwmRectByName,
+                          numMwmIds, MakeNumMwmTree(*numMwmIds, *infoGetter),
+                          traffic::TrafficCache(), index);
+  unique_ptr<WorldGraph> worldGraph = indexRouter.MakeWorldGraph();
+  worldGraph->SetMode(WorldGraph::Mode::SingleMwm);
+
+  // Looking for the best segment for every gate.
+  bool dummy;
+  for (auto & gate : gates)
+  {
+    // Note. For pedestrian routing all the segments are considered as twoway segments so
+    // IndexRouter.FindBestSegment() method finds the same segment for |isOutgoing| == true
+    // and |isOutgoing| == false.
+    Segment bestSegment;
+    if (indexRouter.FindBestSegment(gate.GetPoint(), m2::PointD::Zero() /* direction */,
+                                    true /* isOutgoing */, *worldGraph, bestSegment,
+                                    dummy /* best segment is almost codirectional */))
+    {
+      // Note. There's only one mwm id registered at |numMwmIds|. So numiric mwm id of
+      // |bestSegment| should be zero.
+      CHECK_EQUAL(bestSegment.GetMwmId(), 0,
+                  ("Best segment num mwm id ==", bestSegment.GetMwmId(), ", but should be zero"));
```

Comment:
The comment is redundant because in the case of inequality CHECK_EQUAL will print both arguments.

Reply:
Removed.